packages/native-stack/src/types.tsx

```diff
@@ -596,6 +596,17 @@ export type NativeStackNavigationOptions = {
    * Only supported on iOS and Android.
    */
   freezeOnBlur?: boolean;
+  /**
+   * If a screen is marked `hidden`, the resources will be retained, but it will not be rendered
+   * in the stack. This is used when preloading screen but can also be used as a way to support
+   * iOS picture-in-picture, where you have a video overlay originating on a screen, and even
+   * when popping the screen off the stack the resources need to be retained (not garbage collected),
+   * so that the video can continue playing. Additionally, this allows the screen to be "un-hidden"
+   * (visually, pushed back onto the stack) when the user taps on the overlay video.
+   *
+   * Only supported on iOS and Android.
+   */
+  hiddenFromStack?: boolean;
```

Comment:
Let's not expose this option. We have different plans for how the functionality will be exposed and need to align between native stack and JS https://github.com/orgs/react-navigation/projects/1?pane=issue&itemId=48590954

Reply:
All right, I will remove that. Also I don't have access to this link so I can't check what the plans are.

Reply:
The current idea is to have an action called retain:

1. Let's say you're on a screen where you want to have PiP enabled if the user pops it, you call `navigation.retain(true)`
2. After this, if the screen is removed from the stack with etc., it will still keep it rendered and hidden.
3. And when you navigate back to this screen, it'll be added to the stack and focused
4. Calling `navigation.retain(false)` will disable this behavior, or if the screen is currently hidden, it'll completely remove it

Reply:
Sounds good actually! Thanks for explaining